src/modules/hub.bicep

```diff
@@ -62,49 +65,60 @@ resource defaultTelemetry 'Microsoft.Resources/deployments@2022-09-01' = if (ena
 }
 
 // ADLSv2 storage account for staging and archive
-module storageAccount 'Microsoft.Storage/storageAccounts/deploy.bicep' = {
-  name: 'storage'
-  params: {
-    name: storageAccountName
-    location: location
-    storageAccountSku: storageSku
-    tags: resourceTags
+
+resource storageAccount 'Microsoft.Storage/storageAccounts@2021-08-01' = {
+  name: storageAccountName
+  location: location
+  sku: {
+    name: storageSku
+  }
+  kind: 'BlockBlobStorage'
+  properties: {
+    supportsHttpsTrafficOnly: true
+    isHnsEnabled: true
+    minimumTlsVersion: 'TLS1_2'
+    allowBlobPublicAccess: false
   }
 }
 
-module dataFactory 'Microsoft.DataFactory/factories/deploy.bicep' = {
-  name: 'dataFactory'
-  params: {
-    name: dataFactoryName
-    location: location
-    tags: resourceTags
-  }
+resource blobService 'Microsoft.Storage/storageAccounts/blobServices@2021-06-01' = {
```

Comment:
FYI - You can also nest these within the parent resource if you want to make them a child. Not sure which is preferred. Just mentioning this in case you didn't know. When you nest them, you can leave out the `parent`.

Reply:
I need to do it this way so I can create the containers independently and refer to them by symbollic name when configuring the datasets and trigger.